macros/schema_tests/aggregate_functions/expect_column_most_common_value_to_be_in_set.sql

```diff
@@ -48,7 +52,7 @@ value_counts_ranked as (
 
     select
         *,
-        row_number() over(order by value_count desc) as value_count_rank
+        rank() over(order by value_count desc) as value_count_rank
```

Comment:
Yikes, it just occurred to me that after this, the check for top_n should be 
```sql
value_count_rank <= {{ top_n }}
```
vs
```sql
value_count_rank = {{ top_n }}
```
right? We only seem to test for `top_n: 1` so I don't think I caught this before?

I know this isn't in scope for you for this PR, but that does make sense, or am I misremembering? It's been a while since I worked on this.

Reply:
It seems to me (maybe naively) that if we make this change to `value_count_rank <= {{ top_n }}`, the change from `row_number` to `rank` should be sufficient to deal with collisions?
E.g. in `data_test` the most common value for `col_numeric_a` is `0.5` with 2 counts, with `0` and `1` both having 1 count, thus getting the same rank, `2`. 
So, this test would fail given my suggestion:
```yaml
- dbt_expectations.expect_column_most_common_value_to_be_in_set:
      value_set: [0.5, 1]
      top_n: 2
```
because `0` is just as common but not in the set, correct?

Reply:
I pushed an example to this test PR to help explain what I mean: 
https://github.com/calogica/dbt-expectations/pull/260

Reply:
Indeed, using `rank` and `value_count_rank <= {{ top_n }}`  solves the issue. Should I close this PR and open a feature request for "ties_okay"?

P.S.: Maybe adding following test for "col_numeric_a" might be a good idea to explicitly target top_n and prevent accidental 
```yaml
             # Expect failure if value not in column 
          - dbt_expectations.expect_column_most_common_value_to_be_in_set:
              value_set: [123456789]
              quote_values: false
              top_n: >
                (select count(*) from {{ref('data_test')}})
              config:
                error_if: "=0"
                warn_if: "<>3"
```

Reply:
Feel free to modify this PR, or the other one I posted. 
I'd say we can add `ties_okay` and the change to rank and the `<=` operator all in one PR. Wdyt?

The new test makes sense (if that works with the ref in the yaml?).

Thanks!

Reply:
Works for me! I modified this PR and added the test.